FontoXPathCSharp/Expressions/DataTypes/Casting/CastToBoolean.cs

```diff
@@ -24,14 +24,20 @@ public static CastingFunction ToBoolean(ValueType from)
                     "true" or "1" => new SuccessResult<AtomicValue>(AtomicValue.TrueBoolean),
                     "false" or "0" => new SuccessResult<AtomicValue>(AtomicValue.FalseBoolean),
                     _ => new ErrorResult<AtomicValue>(
-                        "XPTY0004: Casting not supported from given type to xs:boolean or any of its derived types.")
+                        "Casting not supported from given type to xs:boolean or any of its derived types.",
+                        new Error[]
+                        {
+                            new("XPTY0004",
+                                "Casting not supported from given type to xs:boolean or any of its derived types.")
```

Comment:
Why is this duplicated now? Can't we make the ErrorResult accept an XPathError as its argument?